core/shared/src/main/scala/cats/effect/IO.scala

```diff
@@ -1961,6 +1961,17 @@ private object SyncStep {
             }
             .handleErrorWith(t => interpret(f(t), limit - 1))
 
+        case IO.Uncancelable(body, _)
+            if G.isInstanceOf[MonadCancel.Uncancelable[G, Throwable] @unchecked] =>
```

Comment:
Okay, so finally looped around to this. The correct way to do something like this is via `rootCancelScope`:

```suggestion
        case IO.Uncancelable(body, _) if G.rootCancelScope == CancelScope.Uncancelable =>
```

Reply:
Ah, right! Forgot about this.

Reply:
I already made the change locally :-)